PHPStan 0.12+ @implements and @extends annotations

I've encountered two new errors related to PHPStan 0.12 support for generics : 

```                                                                                                                        
 [ERROR] Could not process "test.php"  
         file, due to:                                                                                                  
         "Implement "PHPStan\PhpDocParser\Ast\PhpDoc\ExtendsTagValueNode" to                                            
         "Rector\BetterPhpDocParser\Attributes\Ast\AttributeAwareNodeFactory::createFromPhpDocValueNode"".              
```

```                                                                   
 [ERROR] Could not process "test2.php" file, due  
         to:                                                                                                            
         "Implement "PHPStan\PhpDocParser\Ast\PhpDoc\ImplementsTagValueNode" to                                         
         "Rector\BetterPhpDocParser\Attributes\Ast\AttributeAwareNodeFactory::createFromPhpDocValueNode"".              
```

```php
<?php
# test.php

/**
 * @extends ArrayCollection<int, SomeEntity>
 */
final class SomeEntityCollection extends ArrayCollection
{
    // ...
}
```

```php
<?php
# test2.php

/**
 * @implements IteratorAggregate<int, SomeObject>
 */
final class SomeObjectIterator implements IteratorAggregate
{
    // ...
}
```

--- a/packages/BetterPhpDocParser/src/Attributes/Ast/AttributeAwareNodeFactory.php
+++ b/packages/BetterPhpDocParser/src/Attributes/Ast/AttributeAwareNodeFactory.php
@@ -6,7 +6,9 @@ namespace Rector\BetterPhpDocParser\Attributes\Ast;
 
 use PHPStan\PhpDocParser\Ast\Node;
 use PHPStan\PhpDocParser\Ast\PhpDoc\DeprecatedTagValueNode;
+use PHPStan\PhpDocParser\Ast\PhpDoc\ExtendsTagValueNode;
 use PHPStan\PhpDocParser\Ast\PhpDoc\GenericTagValueNode;
+use PHPStan\PhpDocParser\Ast\PhpDoc\ImplementsTagValueNode;
 use PHPStan\PhpDocParser\Ast\PhpDoc\InvalidTagValueNode;
 use PHPStan\PhpDocParser\Ast\PhpDoc\MethodTagValueNode;
 use PHPStan\PhpDocParser\Ast\PhpDoc\ParamTagValueNode;
@@ -30,7 +32,9 @@ use PHPStan\PhpDocParser\Ast\Type\TypeNode;
 use PHPStan\PhpDocParser\Ast\Type\UnionTypeNode;
 use Rector\BetterPhpDocParser\Ast\PhpDocNodeTraverser;
 use Rector\BetterPhpDocParser\Attributes\Ast\PhpDoc\AttributeAwareDeprecatedTagValueNode;
+use Rector\BetterPhpDocParser\Attributes\Ast\PhpDoc\AttributeAwareExtendsTagValueNode;
 use Rector\BetterPhpDocParser\Attributes\Ast\PhpDoc\AttributeAwareGenericTagValueNode;
+use Rector\BetterPhpDocParser\Attributes\Ast\PhpDoc\AttributeAwareImplementsTagValueNode;
 use Rector\BetterPhpDocParser\Attributes\Ast\PhpDoc\AttributeAwareInvalidTagValueNode;
 use Rector\BetterPhpDocParser\Attributes\Ast\PhpDoc\AttributeAwareMethodTagValueNode;
 use Rector\BetterPhpDocParser\Attributes\Ast\PhpDoc\AttributeAwareParamTagValueNode;
@@ -159,6 +163,16 @@ final class AttributeAwareNodeFactory
 
         if ($phpDocTagValueNode instanceof GenericTagValueNode) {
             return new AttributeAwareGenericTagValueNode($phpDocTagValueNode->value);
+        }
+
+        if ($phpDocTagValueNode instanceof ExtendsTagValueNode) {
+            $typeNode = $this->createFromTypeNode($phpDocTagValueNode->type);
+            return new AttributeAwareExtendsTagValueNode($typeNode, $phpDocTagValueNode->description);
+        }
+
+        if ($phpDocTagValueNode instanceof ImplementsTagValueNode) {
+            $typeNode = $this->createFromTypeNode($phpDocTagValueNode->type);
+            return new AttributeAwareImplementsTagValueNode($typeNode, $phpDocTagValueNode->description);
         }
 
         if ($phpDocTagValueNode instanceof InvalidTagValueNode) {

--- a/packages/BetterPhpDocParser/src/Attributes/Ast/PhpDoc/AttributeAwareExtendsTagValueNode.php
+++ b/packages/BetterPhpDocParser/src/Attributes/Ast/PhpDoc/AttributeAwareExtendsTagValueNode.php
@@ -1,0 +1,14 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\BetterPhpDocParser\Attributes\Ast\PhpDoc;
+
+use PHPStan\PhpDocParser\Ast\PhpDoc\ExtendsTagValueNode;
+use Rector\BetterPhpDocParser\Attributes\Attribute\AttributeTrait;
+use Rector\BetterPhpDocParser\Contract\PhpDocNode\AttributeAwareNodeInterface;
+
+final class AttributeAwareExtendsTagValueNode extends ExtendsTagValueNode implements AttributeAwareNodeInterface
+{
+    use AttributeTrait;
+}

--- a/packages/BetterPhpDocParser/src/Attributes/Ast/PhpDoc/AttributeAwareImplementsTagValueNode.php
+++ b/packages/BetterPhpDocParser/src/Attributes/Ast/PhpDoc/AttributeAwareImplementsTagValueNode.php
@@ -1,0 +1,14 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\BetterPhpDocParser\Attributes\Ast\PhpDoc;
+
+use PHPStan\PhpDocParser\Ast\PhpDoc\ImplementsTagValueNode;
+use Rector\BetterPhpDocParser\Attributes\Attribute\AttributeTrait;
+use Rector\BetterPhpDocParser\Contract\PhpDocNode\AttributeAwareNodeInterface;
+
+final class AttributeAwareImplementsTagValueNode extends ImplementsTagValueNode implements AttributeAwareNodeInterface
+{
+    use AttributeTrait;
+}

